### PR TITLE
Add US keymap and support for Pro S

### PR DESCRIPTION
**Pull request by pokonski**
I own a Masterkeys Pro S keyboard (tenkeyless) and I mapped the US version codes for every key I have on my keyboard, but of course it is missing numpad.

Additionally I added a full layout and ability to switch products via constructor

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 CC=gcc
 CFLAGS=-s -Wall -Wextra -std=c99 -fPIC
-LDFLAGS=-lusb-1.0 -lcmmk -Lout/
+LDFLAGS=-lcmmk -lusb-1.0 -Lout/
```
Comment (pokonski) [inline Makefile:3]: In newer versions of `gcc` apparently the order needs to be like above.

File: libcmmk.h [hunk 2]
```diff
@@ -6,7 +6,7 @@
 #define CMMK_LAYOUT_EU 1 /* keymap_eu.h */
 #define CMMK_LAYOUT_US 2 /* keymap_us.h, XXX: needs outside help for this one */
 
-#define CMMK_LAYOUT CMMK_LAYOUT_EU
+#define CMMK_LAYOUT CMMK_LAYOUT_US
```
Comment (pokonski) [inline libcmmk.h:9]: This probably needs to be switchable

File: libcmmk.c [hunk 3]
```diff
@@ -231,30 +232,56 @@ int cmmk_set_effect_off(struct cmmk *dev)
...
+
+			{K_CAPSLCK, K_A, K_S, K_D, K_F, K_G, K_H, K_J, K_K, K_L, K_SEMICOLON, K_QUOTE, K_ENTER, -1, -1,
+			-1, -1, -1, -1, -1, -1, -1},
 
-	/* Model: MK Pro L [Black/White] EU/Ger */
-	int map[6][22] = {
-		{K_ESC, K_F1, K_F2, K_F3, K_F4, -1, K_F5, K_F6, K_F7, K_F8, -1, K_F9, K_F10, K_F11, K_F12,
-		 K_PRTSCR, K_SCRLCK, K_PAUSE, K_P1, K_P2, K_P3, K_P4},
+			{K_LSHIFT, -1, K_Z, K_X, K_C, K_V, K_B, K_N, K_M, K_COMMA, K_PERIOD, K_FORWARD_SLASH,
+			K_RSHIFT, -1, -1, -1, K_ARROW_UP, -1, -1, -1, -1, -1},
```
Comment (pokonski) [inline libcmmk.c:253]: This part needs filling by someone with full layout US version but for Masterkeys Pro L.

**Comment by chmod222** [comment]
Hey, sorry for the long wait.

I wanted to merge the changes as-is but it struck me how my way of representing the layouts was less than optimal, so I rewrote the thing, so the layout is now passed as a third parameter to `attach()`.

I ditched the old way of representing keys using `K_Keyname` because this will all break down once multiple layouts are in play and would force the library to be compiled separately for every layout. The library now uses the same schema as the official SDK, using the 6x22 matrix only, so keys no longer define the position in the matrix, the matrix defines the position of the keys.

I hope your mapping survived my refactoring. The new `cmmk-test` binary should light up all your keys from top left to bottom right in succession. Just make sure you change the parameter in test.c:29 to `CMMK_LAYOUT_US_S`, as I shamelessly overrode it with my testing version.

**Comment by pokonski** [comment]
Hey! No worries about the refactoring, will have a look and confirm it works today :)

**Comment by pokonski** [comment]
Just tried and all works well!

**Pull request merged**